geometry-builder.cpp

```diff
@@ -760,8 +759,78 @@ void geometry_builder::set_exclude_broken_polygon(int exclude)
     excludepoly = exclude;
 }
 
+void geometry_builder::set_reprojection(struct reprojection *r)
+{
+    reprojection = r;
+}
+
+/** 
+ * Computes area of given polygonal geometry.
+ * Returns 
+ * - the standard GEOS getArea result if area reprojection has not been enabled
+ * - the GEOS getArea resolt of this geometry projected to EPSG:3857 if reprojection has been enabled
+ */
+double geometry_builder::getArea(const geos::geom::Geometry *geom) const
+{
+    // reprojection is not necessary, or has not been asked for.
+    if (!reprojection)
+    {
+        return geom->getArea();
+    }
+
+    // MultiPolygon - return sum of individual areas
+    if (const geos::geom::MultiPolygon* multi = dynamic_cast<const geos::geom::MultiPolygon *>(geom))
+    {
+        double area = 0.0;
+        for (std::size_t i=0; i<multi->getNumGeometries(); i++)
+        {
+            area += getArea(multi->getGeometryN(i));
+        }
+        return area;
+    }
+
+    const geos::geom::Polygon *poly = dynamic_cast<const geos::geom::Polygon *>(geom);
+    if (!poly) {
+        return 0.0;
+    }
+
+    // standard polygon - reproject rings individually, then assemble polygon and
+    // compute area.
```

Comment:
Thing is we can only reproject individual coordinate pairs, and the wrapper I built will dissect a polygon into rings, the rings into coordinate pairs, reproject them, and reassemble rings and a polygon from that. The wrapper does not have the capability to reproject a multipolygon; this would require iterating through the getGeometryNs of the multipolygon, reprojecting the individual members, and then calling the GEOS factory to build a multipolygon from all that in the end. It would then be possible to call getArea on the multipolygon. But the building of the multipolygon would IMHO be unnecessary extra code and extra memory allocation/release.